chronicles/log_output.nim

```diff
@@ -422,40 +415,87 @@ macro append*(o: var AnyOutput,
   result.add newCall("append", o, arg2)
   for arg in restArgs: result.add newCall("append", o, arg)
 
-proc rfcTimestamp: string =
-  now().format("yyyy-MM-dd HH:mm:ss'.'fffzzz")
+import
+  stew/objects
+
+let timezone = toArray(6, now().format("zzz"))
+const rfcFormat = initTimeFormat "yyyy-MM-dd HH:mm:ss'.'fffzzz"
+
+template rfcTimestamp: string =
+  now().format(rfcFormat)
 
-proc epochTimestamp: string =
+template epochTimestamp: string =
   formatFloat(epochTime(), ffDecimal, 6)
 
-template timestamp(record): string =
-  when record.timestamps == RfcTime:
+template writeTs*(record) =
```

Comment:
the way to implement this efficiently is to cache the timezone offset (in seconds), then offset the incoming time accordingly, and print the offset time as if it were utc and finally making sure the offset is printed as well - this makes it correct, efficient and generic to all time formats